features/prog.go

```diff
@@ -108,7 +108,15 @@ var haveProgramTypeMatrix = internal.FeatureMatrix[ebpf.ProgramType]{
 		},
 	},
 	// ebpf.Extension:             {Version: "5.6"},
-	// ebpf.LSM:                   {Version: "5.7"},
+	ebpf.LSM: {Version: "5.7",
+		Fn: func() error {
+			return probeProgram((&ebpf.ProgramSpec{
+				Type:       ebpf.LSM,
+				AttachType: ebpf.AttachLSMMac,
+				AttachTo:   "file_mprotect",
```

Comment:
How did you chose file_mprotect vs. any other function to attach to?

Reply:
I primarily chose it because we also use it in the tests in this repo
https://github.com/cilium/ebpf/blob/f17b01bb144a74559093d1699f5e03fac18070e8/link/tracing_test.go#L103

And I didn't see any problems with the choice in general because it is generic LSM hook and is not bounded by any special flags. 
Ref https://elixir.bootlin.com/linux/latest/source/include/linux/security.h#L974